File: 2116-count-number-of-pairs-with-absolute-difference-k/count-number-of-pairs-with-absolute-difference-k.java
class Solution {
    public int countKDifference(int[] nums, int k) {
        
        int[] freq = new int[101+k];
        int n = nums.length;
        int pairs = 0;

        for(int i : nums){
            freq[i]++;
        }

        for(int i = 0; i<n; i++){
            pairs += freq[nums[i] + k];
        }

        return pairs;
    }
}

File: 2116-count-number-of-pairs-with-absolute-difference-k/count-number-of-pairs-with-absolute-difference-k.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */

var countKDifference = function(nums, k) {
    let freq = new Array(101).fill(0);
    let pairs = 0;

    for (let num of nums) {
        // count how many numbers already seen can pair with current num
        pairs += (num - k >= 0 ? freq[num - k] : 0);
        pairs += (num + k <= 100 ? freq[num + k] : 0);

        freq[num]++;
    }

    return pairs;
};
